### PR TITLE
docs(forms): properly convert number to string for formControlName input

### PR DESCRIPTION
**Pull request by piotrtomiak**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Currently documentation https://angular.io/api/forms/FormArrayName for `formArrayName` encourages to pass `number` to a `string` input.

## What is the new behavior?

Documentation will encourage to use `i.toString()` to cast index to `string`.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by piotrtomiak** [comment]
Any chance to have this addressed in a near future? I would love to close my issue and let user know, which syntax he should be using.

**Comment by piotrtomiak** [comment]
@jenniferfell @brandonroberts Could I get some traction on this?

**Comment by trotyl** [comment]
I think interpolation here:

```html
<input formControlName="{{i}}" placeholder="City">
```

Might be better than `.toString()` call:

```html
<input [formControlName]="i.toString()" placeholder="City">
```

**Comment by brandonroberts** [comment]
@piotrtomiak I agree with @trotyl, that interpolation for the string would be better here.

File: packages/examples/forms/ts/nestedFormArray/nested_form_array_example.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ import {FormArray, FormControl, FormGroup} from '@angular/forms';
     <form [formGroup]="form" (ngSubmit)="onSubmit()">
       <div formArrayName="cities">
         <div *ngFor="let city of cities.controls; index as i">
-          <input [formControlName]="i" placeholder="City">
+          <input [formControlName]="i.toString()" placeholder="City">
```
Comment (brandonroberts) [inline packages/examples/forms/ts/nestedFormArray/nested_form_array_example.ts]:
```suggestion
          <input [formControlName]="{{i}}" placeholder="City">
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: packages/examples/forms/ts/nestedFormArray/nested_form_array_example.ts (hunk 1 above)
Comment (trotyl) [reply packages/examples/forms/ts/nestedFormArray/nested_form_array_example.ts]: Also no more brackets.

Comment (piotrtomiak) [reply packages/examples/forms/ts/nestedFormArray/nested_form_array_example.ts]: Fixed with force push.

**Comment by piotrtomiak** [comment]
@trotyl Thanks for your suggestion! It indeed looks better.

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Caretaker note: The Codefresh failure is unrelated

**Comment by piotrtomiak** [comment]
@brandonroberts @alxhub Thank you for merging the change!

**Pull request closed without merging**